src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvokerProvider.cs

```diff
@@ -210,8 +207,11 @@ private InnerCache CurrentCache
         internal List<Func<IRazorPage>> GetViewStartFactories(CompiledPageActionDescriptor descriptor)
         {
             var viewStartFactories = new List<Func<IRazorPage>>();
+            // Always pick up all _ViewStarts, including the ones outside the Pages root. This mirrors the
+            // behavior of _ViewImports.cshtml discovery.
```

Comment:
Ahh, there's another overload that does not take `basePath` (which is what we use in the view engine). I'll switch to it.